resources/js/components/dashboard/components/IndividualSponsor.tsx

```diff
@@ -0,0 +1,351 @@
+    private nextSponsor = () => {
+        this.setState({ loading: true });
+        const total = this.state.totalNo;
+        const currentUrl = this.props.history.location.pathname;
+        const base = currentUrl.substring(0, currentUrl.lastIndexOf('/'));
+        const currentSpons:string = currentUrl.substring(currentUrl.lastIndexOf('/')+1);
+        const nextSponsor = (currentSpons != null && !(currentSpons==="")) ? (parseInt(currentSpons) + 1)%total : parseInt(currentSpons)
+        this.props.history.push(`${base}/${nextSponsor}`);
+    }
+
+    private renderSponsor = () => {
+        const {Sponsor,user,resources} = this.state;
+        var loading = 
+            <div style={{textAlign:"center",marginTop:"3em"}}>
+                <Card sectioned><Heading>Loading more info.....</Heading></Card>
+                <div style={{marginTop:"1.5em"}}>
+                    <Spinner color="teal" /> 
+                </div>
+            </div>
+        if(Sponsor) {
+            const tier = Sponsor.tier;
+            const data = this.state.portalInfo.data;
+            const portalInfoImages = this.state.portalInfo.files.filter((x:IAssetInformation)=>{return !x.name.toLowerCase().includes("logo")});
+            const app: ISponsor | undefined = Sponsor;
+            const tier_badge = () => {
+                switch(tier.toLowerCase()) {
+                    case "co-host":
+                        return "info"
+                    case "tera":
+                        return "success"
+                    case "giga":
+                        return "attention"
+                    case "mega":
+                        return "warning"
+                    case "kilo":
+                        return undefined
+                    default: return undefined
+                }
+            }
+            const metadata = <>
+                {<Badge status={tier_badge()}>{this.capitalizeFirstLetter(tier)}</Badge>}
+            </>;
+            const carouselDivs = portalInfoImages.map((p:IAssetInformation)=>{
+                return(
+                    <div>
+                        <img src={p.url} />
+                        <p className="legend">{p.name}</p>
+                    </div>
+                )
+            })
+            const placeholderCarousel = (
+                <div>
+                    <img style={{width:"100%"}} id="image" src="https://s3.eu-west-2.amazonaws.com/hc-upload/sponsors/not-marhsll-wace/843eb30a-011e-4c42-a1c1-bde699b27f1d.jpg"></img>
+                    <p className="legend">An office</p>
+                </div>
+            )
+            let logoUrl: string | undefined = this.state.portalInfo.files.find((x:IAssetInformation)=> {return x.name.toLowerCase().includes("logo")});
+            if (!logoUrl || logoUrl === undefined) {
+                // Use backup logo
+                logoUrl = "https://www.pngfind.com/pngs/m/665-6659827_enterprise-comments-default-company-logo-png-transparent-png.png";
```

Comment:
Not sure if we can link other people's photos in the website (I think we need to ask them for consent):

Can you download the photo and we will add it to images and load it from there?

Reply:
Done